mesos-rxjava-example/mesos-rxjava-example-framework/src/main/java/org/apache/mesos/rx/java/example/framework/sleepy/State.java

```diff
@@ -6,18 +6,30 @@
 import org.slf4j.LoggerFactory;
 
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import static com.google.common.collect.Maps.newConcurrentMap;
 
 final class State {
     private static final Logger LOGGER = LoggerFactory.getLogger(State.class);
 
-    private Protos.FrameworkID fwId;
+    private final double cpusPerTask;
+    private final double memMbPerTask;
 
     @NotNull
     private final Map<Protos.TaskID, Protos.TaskState> taskStates;
+    @NotNull
+    private final AtomicInteger offerCounter = new AtomicInteger();
+    @NotNull
+    private final AtomicInteger totalTaskCounter = new AtomicInteger();
```

Comment:
Rx doesn't make any guarantees about serial processing that's delegated to the scheduler that is configured for observation and subscription. So there is the possibility that things will be accessed from multiple threads at the same time.

I'll remove the unnecessary synchronize annotations, but I'll keep the atomic ints.